lib/main.dart

```diff
@@ -9,11 +13,66 @@ import 'package:seeds/screens/app/app.dart';
 import 'package:seeds/screens/onboarding/onboarding.dart';
 import 'package:seeds/widgets/passcode.dart';
 import 'package:seeds/widgets/splash_screen.dart';
+import 'package:sentry/sentry.dart' as Sentry;
 
 import 'generated/r.dart';
 
+final Sentry.SentryClient _sentry = Sentry.SentryClient(
+    dsn: "https://ee2dd9f706974248b5b4a10850586d94@sentry.io/2239437");
+
+bool get isInDebugMode {
+  bool inDebugMode = false;
+  assert(inDebugMode = true);
+  return inDebugMode;
+}
+
+/// Reports [error] along with its [stackTrace] to Sentry.io.
+Future<Null> _reportError(dynamic error, dynamic stackTrace) async {
+  print('Caught error: $error');
+  print('Reporting to Sentry.io...');
+
+  final Sentry.SentryResponse response = await _sentry.captureException(
+    exception: error,
+    stackTrace: stackTrace,
+  );
+
+  if (response.isSuccessful) {
+    print('Success! Event ID: ${response.eventId}');
+  } else {
+    print('Failed to report to Sentry.io: ${response.error}');
+  }
+}
+
 main(List<String> args) async {
-  runApp(SeedsApp());
+  WidgetsFlutterBinding.ensureInitialized();
+  SystemChrome.setPreferredOrientations(
+      [DeviceOrientation.portraitUp, DeviceOrientation.portraitDown]).then((_) {
+    if (isInDebugMode) {
+      runApp(SeedsApp());
+    } else {
+      FlutterError.onError = (FlutterErrorDetails details) async {
+        print('FlutterError.onError caught an error');
+        await _reportError(details.exception, details.stack);
+      };
+
+      Isolate.current.addErrorListener(
+        RawReceivePort((dynamic pair) async {
+          print('Isolate.current.addErrorListener caught an error');
+          await _reportError(
+            (pair as List<String>).first,
+            (pair as List<String>).last,
+          );
+        }).sendPort,
+      );
+
+      runZoned<Future<Null>>(() async {
+        runApp(SeedsApp());
```

Comment:
Most important question - how exactly we need to change services code to throw proper errors given this handling?

Reply:
We just need to make sure that the code throws error and all the other things will be taken care by sentry.